crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -209,29 +216,82 @@ impl CairoHintProcessor<'_> {
                 )?)?);
 
                 self.cheated_state
-                    .start_prank(contract_address, caller_address)
+                    .start_prank(contract_address, caller_address);
+                Ok(())
             }
             "stop_prank" => {
                 let contract_address = ContractAddress(PatriciaKey::try_from(StarkFelt::new(
                     inputs[0].clone().to_be_bytes(),
                 )?)?);
 
-                self.cheated_state.stop_prank(contract_address)
+                self.cheated_state.stop_prank(contract_address);
+                Ok(())
             }
             "mock_call" => todo!(),
-            "declare" => self.cheated_state.declare(
-                &mut buffer,
-                &mut self.blockifier_state,
-                &inputs,
-                // TODO(#41) Remove after we have a separate scarb package
-                &contracts
-                    .iter()
-                    .map(|(k, v)| (k.clone(), ContractArtifacts::from(v)))
-                    .collect(),
-            ),
-            "deploy" => self
-                .cheated_state
-                .deploy(&mut buffer, &mut self.blockifier_state, &inputs),
+            "declare" => {
+                let contract_name = inputs[0].clone();
+
+                match self.cheated_state.declare(
+                    &mut self.blockifier_state,
+                    &contract_name,
+                    // TODO(#41) Remove after we have a separate scarb package
+                    &contracts
+                        .iter()
+                        .map(|(k, v)| (k.clone(), ContractArtifacts::from(v)))
+                        .collect(),
+                ) {
+                    Ok(class_hash) => {
+                        let felt_class_hash =
+                            felt252_from_hex_string(&class_hash.to_string()).unwrap();
+
+                        buffer
+                            .write(Felt252::from(0))
+                            .expect("Failed to insert error code");
+                        buffer
+                            .write(felt_class_hash)
+                            .expect("Failed to insert declared contract class hash");
+                        Ok(())
+                    }
+                    Err(CheatcodeError::Recoverable(_)) => {
+                        panic!("Declare should not fail recoverably!")
+                    }
+                    Err(CheatcodeError::Unrecoverable(err)) => Err(err),
+                }
+            }
+            "deploy" => {
+                // TODO(#1991) deploy should fail if contract address provided doesn't match calculated
```

Comment:
Are you sure about this number? 😅

Reply:
It's an old protostar issue

Reply:
Hmm, I thought that all protostar issues were ported here

Reply:
yes, but this one was closed without addressing this Todo

Reply:
I think this TODO is no relevant anymore